### PR TITLE
#10185 add missing database tables to the database export view

### LINKED ISSUE #10185: Some Database Tables are missing in the Database Export View [2]
### Bug Description

There are 69 database tables defined in the [DatabaseTable.java](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/sormas-api/src/main/java/de/symeda/sormas/api/importexport/DatabaseTable.java), those are available for export in the SORMAS UI. However, there are more than 69 (non history) tables in the SORMAS database, some of which (maybe even all) should be downloadable.

Missing are the tables

- campaign_campaignformdata (is already added)
- person_locations
- task_observer

The following tables are also missing and should be added:
- contacts_visits
- events_eventgroups
- campaign_campaignformmeta
- sharerequestinfo_shareinfo

### Steps to Reproduce

1. Try to download the aforementioned missing tables via the UI
-> it is not possible

### Expected Behavior

It was stated by HZI that some (or even all) of the missing tables should be downloadable via the UI [TBD]

### Acceptance Criteria
Extend the tests to check if all tables of this kind are exported

### System Details

* Device:
* SORMAS version: 1.75.0-SNAPSHOT
* Android version/Browser: Any browser
* Server URL: Any system
* User Role:

### Additional Information

This issue came up as a side note in a discussion with HZI. I don't know how urgent or important this is.

Not all tables defined in the [DatabaseTable.java](https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/sormas-api/src/main/java/de/symeda/sormas/api/importexport/DatabaseTable.java) are always downloadable via the SORMAS UI. Some are hidden when the related feature is disabled in the feature configuration.

### PR DESCRIPTION
**Pull request by carina29**
Fixes #10185

### CODE REVIEW

**Review by leventegal-she — CHANGES_REQUESTED** [review]
Please extend the test `DatabaseExportServiceTest::test_all_entities_have_export_configuration` to check the join tables to have export configuration.

File: sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java [hunk 1]
```diff
@@ -62,6 +64,10 @@ public void testGetConfigFullyDefined() {
 	@Test
 	public void test_all_entities_have_export_configuration() {
 		Collection<String> exportableTables = DatabaseExportService.EXPORT_CONFIGS.values();
+		Set<DatabaseTable> joinTables = Arrays.stream(DatabaseTable.values()).filter(t -> t.getParentTable() != null).collect(Collectors.toSet());
+		Set<DatabaseTable> joinTablesFromExportConfigs =
+			DatabaseExportService.EXPORT_CONFIGS.keySet().stream().filter(t -> t.getParentTable() != null).collect(Collectors.toSet());
+
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java:70]: This just tests that all items from DatabaseTable has an export config.

this tes checks the table names of entity classes, you should do similarly for join tables but the table names you get from  @JoinTable annotations of the getter methods

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java (hunk 1 above)
Comment (carina29) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java:70]: Ok, sure. I've understood something else last time ( what I've implemented above)

File: sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java [hunk 2]
```diff
@@ -78,18 +93,40 @@ public void test_all_entities_have_export_configuration() {
 				if (!exportableTables.contains(tableName)) {
 					missingEntities.add(clazz.getSimpleName());
 				} else if (NOT_EXPORTED_ENTITIES.contains(clazz.reflect())) {
-					exportedButNotWanted.add(clazz.getSimpleName());
+					exportedButNotWantedEntity.add(clazz.getSimpleName());
+				}
+			}
+
+			for (JavaMethod method : clazz.getMethods()) {
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java:100]: this for cycle should go inside the if above.

There is no need to search for @JoinTable in not entity classes

File: sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java [hunk 3]
```diff
@@ -59,11 +65,20 @@ public void testGetConfigFullyDefined() {
 		FirstVaccinationDate.class,
 		AuditLogEntry.class);
 
+	private static final List<String> NOT_EXPORTED_JOIN_TABLE_NAMES = Arrays.asList(
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java:68]: please create a ticket to don't forget to add these tables too to the database export. Or just do it now :)

**Review by leventegal-she — CHANGES_REQUESTED** [review]

File: sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java (hunk 3 above)
Comment (carina29) [reply sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java]: I've added into the ticket description that these 4 tables should also be added: 
contacts_visits
events_eventgroups
campaign_campaignformmeta
sharerequestinfo_shareinfo

https://github.com/hzi-braunschweig/SORMAS-Project/issues/10185

File: sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java [hunk 4]
```diff
@@ -78,7 +82,21 @@ public void test_all_entities_have_export_configuration() {
...
-					exportedButNotWanted.add(clazz.getSimpleName());
+					exportedButNotWantedEntity.add(clazz.getSimpleName());
+				}
+
+				for (JavaMethod method : clazz.getMethods()) {
+					if (method.isAnnotatedWith(JoinTable.class)) {
+						JoinTable joinTableAnnotation = method.getAnnotationOfType(JoinTable.class);
+						String joinTableName = joinTableAnnotation.name();
+						if (StringUtils.isBlank(joinTableName)) {
+							joinTableName = clazz.getSimpleName().toLowerCase();
```
Comment (leventegal-she) [inline sormas-backend/src/test/java/de/symeda/sormas/backend/importexport/DatabaseExportServiceTest.java:93]: the name of the join table should not be the name of the class.  Pleas let the test fail if there is no joinTableName

**Review by leventegal-she — CHANGES_REQUESTED** [review]

**Review by leventegal-she — APPROVED** [review]

**Pull request merged**